test/core/credentials/transport/tls/tls_security_connector_test.cc

```diff
@@ -140,13 +155,14 @@ TEST_F(TlsSecurityConnectorTest,
   TlsChannelSecurityConnector* tls_connector =
       static_cast<TlsChannelSecurityConnector*>(connector.get());
   EXPECT_NE(tls_connector->ClientHandshakerFactoryForTesting(), nullptr);
-  EXPECT_EQ(tls_connector->RootCertsForTesting(), root_cert_0_);
+  auto root = tls_connector->RootCertInfoForTesting();
```

Comment:
I don't think we actually need the changes here.  We should be able to stick with the original assertion, which now would compare the shared_ptr instances by pointer value, which should be the same.

Same throughout.